src/unix/core.c

```diff
@@ -845,22 +846,23 @@ void uv__io_init(uv__io_t* w, uv__io_cb cb, int fd) {
 }
 
 
-void uv__io_start(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
+int uv__io_start(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   assert(0 == (events & ~(POLLIN | POLLOUT | UV__POLLRDHUP | UV__POLLPRI)));
   assert(0 != events);
   assert(w->fd >= 0);
   assert(w->fd < INT_MAX);
 
   w->pevents |= events;
-  maybe_resize(loop, w->fd + 1);
+  if (maybe_resize(loop, w->fd + 1))
```

Comment:
I think this must go too, otherwise the state will get very unpredictable in other places.

Reply:
At the very least the `w->pevents |= events` should be done _after_ the `maybe_resize()` call.  And yes, returning errors means all callers need to start handling them, otherwise there is no point.